roboteq_diff_driver/src/driver.cpp

```diff
@@ -655,17 +236,39 @@ void MainNode::odom_loop()
           if (odom_buf[delim] == ':')
           {
             odom_buf[delim] = 0;
-            odom_encoder_right = (int32_t)strtol(odom_buf+3, NULL, 10);
-            odom_encoder_left = (int32_t)strtol(odom_buf+delim+1, NULL, 10);
-#ifdef _ODOM_DEBUG
-ROS_DEBUG_STREAM("encoder right: " << odom_encoder_right << " left: " << odom_encoder_left);
-#endif
+            /* the query in line 200 asks for information every 25ms, hence the 0.025
+            * the factor 4 is a scaling factor of unknown origin, probably has to do something with cpr-ppr
```

Comment:
4 is because of ticks vs. quadrature states. It should be that `cpr = 4*ppr` . Are you sure that this way of computing the rpm is correct? Because we have always observed that `rpm = (clicks / cpr) * 1/dt * 60` which seems to differ by a factor of 1/4. We need to figure out why this has apparently changed?